magellan-library/src/main/java/com/wealthfront/magellan/lifecycle/LifecycleAwareComponent.kt

```diff
@@ -5,6 +5,7 @@ import android.content.Context
 public abstract class LifecycleAwareComponent : LifecycleAware, LifecycleOwner {
 
   private val lifecycleRegistry = LifecycleRegistry()
+  protected val lifecycleLimiter: LifecycleLimiter = LifecycleLimiter(lifecycleRegistry)
```

Comment:
Another idea: if we only want to expose some of the functionality of the `LifecycleRegistry` and not the whole class, what if we made `LifecycleLimiter` an interface and exposed it here under that interface?